sources/views/contacts/contactDescription/activitiesAndFiles.js

```diff
@@ -0,0 +1,213 @@
+										}
+										return false;
+									}
+								}
+							],
+							format: webix.i18n.longDateFormatStr,
+							sort: "date"
+						},
+						{
+							id: "Details",
+							header: ["Details", {content: "textFilter"}],
+							fillspace: true,
+							sort: "string"
+						},
+						{id: "edit", header: "", template: "{common.editIcon()}", width: 50},
+						{id: "del", header: "", template: "{common.trashIcon()}", width: 50}
+					],
+					onClick: {
+						"wxi-trash": (e, id) => this.deleteActivity(id),
+						"wxi-pencil": (e, id) => this.editActivity(id)
+					}
+				},
+				{
+					cols: [
+						{},
+						{
+							view: "button",
+							width: 200,
+							css: "webix_primary",
+							value: "<span class='webix_icon mdi mdi-plus-box'></span> Add Activity",
+							click: () => this.window.showWindow()
+						}
+					]
+				}
+
+			]
+		};
+
+		const filesTable = {
+			view: "form",
+			localId: "files",
+			rows: [
+				{
+					view: "datatable",
+					localId: "filesTable",
+					type: "uploader",
+					borderless: true,
+					columns: [
+						{
+							id: "name",
+							header: "Name",
+							fillspace: true,
+							sort: "string"
+						},
+						{
+							id: "time",
+							header: "Date",
+							template: () => {
+								const format = webix.i18n.longDateFormatStr;
+								return format(new Date());
+							},
+							fillspace: true,
+							sort: "date"
+						},
+						{id: "size", header: "Size", sort: this.sortForSize},
+						{
+							id: "Delete",
+							header: "",
+							template: "<span class='webix_icon wxi-trash'></span>",
+							width: 40
+						}
+					],
+					onClick: {
+						"wxi-trash": (e, id) => this.deleteFile(id)
+					}
+				},
+				{
+					cols: [
+						{},
+						{
+							view: "uploader",
+							width: 200,
+							localId: "upload",
+							value: "Upload file",
+							name: "files",
+							autosend: false
+						}
+					]
+				}
+			]
+		};
+
+		return {
+			rows: [
+				{
+					view: "tabbar",
+					localId: "tab",
+					value: "Activities",
+					options: [
+						{value: "Activities"},
+						{value: "Files"}
+					]
+				},
+				{
+					cells: [
+						{localId: "Activities", rows: [activities]},
+						{localId: "Files", rows: [filesTable]}
+					]
+				}
+			]
+		};
+	}
+
+	init() {
+		this.$$("activitiesTable").sync(dataActivities);
+		this.$$("filesTable").sync(contactFiles);
+		this.window = this.ui(formActivity);
+		this.$$("tab").attachEvent("onChange", id => this.$$(id).show());
+		this.$$("upload").attachEvent("onBeforeFileAdd", (file) => {
+			const idToNumber = +ItemDataContact.getId();
+			const dataFile = {
+				ContactID: idToNumber,
+				name: file.name,
+				date: file.file.lastModifiedDate,
+				size: file.sizetext
+			};
+			contactFiles.add(dataFile);
+		});
+	}
+
+	urlChange() {
+		webix.promise.all([
+			dataActivities.waitData,
+			contactFiles.waitData
+		]).then(() => {
+			const idToNumber = +ItemDataContact.getId();
```

Comment:
Here and below (and in all similar use-cases) convert all IDs to strigns.
id of an item is a parameter defined by backend and will not always be a number in real life.
In addition to it, Webix sometimes may give you numbers as strings.